### ISSUE TITLE
[BUG]Not able to Authenticate with the API request. Cookies are saved but not origins/localStorage

### ISSUE THREAD

**Issue by CkumarGitHub** [microsoft/playwright#26973]
### System info
- Playwright Version: [v1.36]
- Operating System: [macOS 13.2]
- Browser: [All, Chromium, Firefox, WebKit]

**Steps**
- I have created a setup.ts file. Calling 5 APIs to log in and get to the homepage
- Saving the authenticated state
- Running UI tests using storage state
- [...]

**Expected**

The page should load with an authenticated state

**Actual**

The page is not loading with an authenticated state

If in setup I use page. goto(homepage URL) and then save the auth state Then it works but request. storage is only storing cookies

I also checked the auth.json file and only cookies get saved and not the origins/localStorage

**Comment by dgozman** [comment]
We need more information to act on this issue. Could you please provide a repro that we can run locally?

**Comment by CkumarGitHub** [comment]
@dgozman :
The code I cannot share due to company privacy. I tried to find some public site which have exposed the login API but couldn't find it.
So I just wanted to confirm that if await request.storageState({path: 'path'}) saves the origins and localStorage ?. If yes then How does it store as APIContext can't access application->Storage
Because await page.context().storageState({path: 'path'}) does.

**Comment by dgozman** [comment]
@CkumarGitHub If you are using `APIRequestContext` to sign in, it does not save the local storage. If you need local storage for your auth to kick in, you should sign in with a `Page` instead. Does that answer your question?

**Comment by CkumarGitHub** [comment]
@dgozman Yes I need local storage for auth to kick in. Let me go ahead with page.
I will close this ticket in that case
